cli/azd/cmd/deploy.go

```diff
@@ -217,64 +224,72 @@ func (da *deployAction) Run(ctx context.Context) (*actions.ActionResult, error)
 	deployResults := map[string]*project.ServiceDeployResult{}
 
 	for _, svc := range da.projectConfig.GetServicesStable() {
-		stepMessage := fmt.Sprintf("Deploying service %s", svc.Name)
-
-		// Skip this service if both cases are true:
-		// 1. The user specified a service name
-		// 2. This service is not the one the user specified
-		if targetServiceName != "" && targetServiceName != svc.Name {
-			continue
-		}
+		var deployResult *project.ServiceDeployResult
+		operationMessage := fmt.Sprintf("Deploying service %s", svc.Name)
+		err = da.operationManager.Run(ctx, operationMessage, func(operation *operations.Operation) error {
```

Comment:
Note that no where in the command are we orchestrating when to show/stop/update spinners.